frontend/packages/multicluster-sdk/src/internal/fleetK8sWatchResourceStore.ts

```diff
@@ -0,0 +1,309 @@
+/* Copyright Contributors to the Open Cluster Management project */
```

Comment:
Made an independent module to house caching methods, as per general Zustand guidelines.